homeassistant/components/iotty/switch.py

```diff
@@ -138,3 +163,58 @@ def _handle_coordinator_update(self) -> None:
         if isinstance(device, LightSwitch):
             self._iotty_device.is_on = device.is_on
         self.async_write_ha_state()
+
+
+class IottyOutlet(IottyEntity, SwitchEntity):
```

Comment:
this is VERY similar to the `IottyLightSwitch` leading to lots of duplicate code. If you allow passing in an `entity_description` for things like the `device_class`  (actually only the device class would already do, but I'd say plan ahead and use a `SensorEntityDescription`) and make use of generic for the `iotty_device` type, it looks to me like you can do with minimal code changes.

Reply:
I see your point. However, we're planning some additional changes that would further differentiate the two.
I also believe in the spirit of OSS and the contribution style of HomeAssistant such a distinction would make it easier to understand and review the code in the future (not to mention, it would be easier for any user to read it by themselves and tinker with it as they wish, which I believe is one of the advantages of HA).

This is the rationale behind my design choice. Let me know what you think about that!
In case you're not convinced, this is the attribute you're talking about, right? https://developers.home-assistant.io/docs/core/entity?_highlight=entity_description#entity-description

If that is the case, you're suggesting I store a `SwitchDeviceClass` in it, and then use conditional statements in every place where behaviour is different to decide which code to execute?

Reply:
Duplicate code is almost never a good idea. Yes, those entity descriptions. Not conditionals, but check other integrations e.g. lamarzocco for how entity descriptions can be extended with Callables to allow for different requirements of diferent switches. If actual implementations differ too much, at least a  common base class is often a good idea.

Reply:
That makes sense! I just finished setting up the code so it looks more like the scenario you described. Let me know if that works for you!